### PR TITLE
feat: [CI-6562]: run pipeline modal for v1 pipeline repository changes

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

As part of Harness YAML simplification process that's currently going on, we've added new run pipeline modal for V1 Pipeline, pertaining to follow high level changes:

1. This PR includes changes to open new run pipeline modal with just codebase related changes
2. Repo is a first class element in new simplified yaml and values will be fetched from new inputs api
3.. All these changes are being controlled by a feature flag CI_YAML_VERSIONING.

#### Screenshots

Screenshot 2023-02-23 at 11 27 47 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      })
+      const existingValues = { ...formik?.values }
+      const updatedValues = set(existingValues, codeBaseTypePath, codeBaseType)
+      formik?.setValues(updatedValues)
+    }
+  }, [codeBaseType])
+
+  const handleTypeChange = (newType: CodebaseTypes): void => {
+    formik?.setFieldValue(`repository.reference.type`, '')
+    formik?.setFieldValue(codeBaseTypePath, newType)
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:112]: @shaurya-harness the version of formik we're currently using does not guarantee batching above two `setFieldValue` calls. Instead use `setValues` from formik. Docs with more details https://docs.google.com/document/d/11QYKW7lDtVQ_gUUQvCeEB6v3CsZz5g_yHidcspgdXs0/edit#heading=h.5hyrjy2fma40

**Review by vardanbansal-harness — CHANGES_REQUESTED** [review]
Pls take care of the comment related to formik below.

**Comment by shaurya-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/RunPipelineFormV1.tsx [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+          onContinuePipelineClick={() => {
+            hidePreflightCheckModal()
+            handleRunPipeline(valuesPipelineRef.current, true)
+          }}
+        />
+      </Dialog>
+    )
+  }, [notifyOnlyMe])
+
+  useEffect(() => {
```
Comment (d1wilko) [inline src/modules/70-pipeline/v1/components/RunPipelineModalV1/RunPipelineFormV1.tsx:257]: move higher in file :)

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/useRunPipelineModalV1.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        </Layout.Vertical>
+      </Dialog>
+    ),
+    [branch, repoIdentifier, pipelineIdentifier, inputSetSelected, planExecutionId]
+  )
+
+  const open = useCallback(() => {
+    // if (planExecutionId) {
+    //   fetchExecutionData()
+    // }
```
Comment (d1wilko) [inline src/modules/70-pipeline/v1/components/RunPipelineModalV1/useRunPipelineModalV1.tsx:110]: can we get rid of this?

**Review by d1wilko — CHANGES_REQUESTED** [review]
Coming together nicely - added some things to consider

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 4]
```diff
@@ -384,7 +396,13 @@ export const MenuCell: CellType = ({ row, column }) => {
         canEdit={canEdit}
         onViewCompiledYaml={() => onViewCompiledYaml(data)}
         onCompareExecutions={() => addToCompare(data)}
-        onReRunInDebugMode={hasCI && CI_REMOTE_DEBUG ? () => openRunPipelineModal() : undefined}
+        onReRunInDebugMode={
+          hasCI && CI_REMOTE_DEBUG
+            ? CI_YAML_VERSIONING && module?.valueOf().toLowerCase() === moduleToModuleNameMapping.ci.toLowerCase()
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:401]: Have added this util but might take time to get it merged. We can clean this up after that happens.

File: src/modules/70-pipeline/v1/components/PipelineCanvasV1/PipelineCanvasV1.tsx [hunk 5]
```diff
@@ -433,23 +429,23 @@ export function PipelineCanvasV1({
...
+  //         gitDetails: {
+  //           repoIdentifier: inputSetRepoIdentifier,
+  //           branch: inputSetBranch
+  //         }
+  //       }
+  //     ]
+  //     return inputSetSelected
+  //   }
+  //   return []
+  // }
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/v1/components/PipelineCanvasV1/PipelineCanvasV1.tsx:448]: +1

[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/PipelineCanvasV1/PipelineCanvasV1.tsx:448]: @vardanbansal-harness  this is your code, will delete this but we might need to add it back when run modal supports selection of old input sets

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+    }
+  }, [loadingConnectorDetails, connectorDetails])
+
+  const shouldAllowRepoFetch = useMemo((): boolean => {
+    return (
+      viewType === StepViewType.DeploymentForm &&
+      codeBaseType === CodebaseTypes.BRANCH &&
+      !get(formik?.values, codeBaseInputFieldFormName) &&
+      !isDefaultBranchSet &&
+      !isUndefined(codebaseConnector)
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:207]: +1

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+          accountIdentifier: accountId,
+          orgIdentifier: connectorScope === Scope.ORG || connectorScope === Scope.PROJECT ? orgIdentifier : undefined,
+          projectIdentifier: connectorScope === Scope.PROJECT ? projectIdentifier : undefined
+        }
+      })
+    }
+  }, [connectorId])
+
+  useEffect(() => {
+    if (!loadingConnectorDetails && !isUndefined(connectorDetails)) {
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:196]: or just `connectorDetails` with it's truthy value should be good enough

File: src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx [hunk 8]
```diff
@@ -76,7 +79,7 @@ export interface ExecutionActionsProps {
   source: ExecutionPathProps['source']
   onViewCompiledYaml?: () => void
   onCompareExecutions?: () => void
-  onReRunInDebugMode?: () => void
+  onReRunInDebugMode?: any
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx:82]: avoid any, it becomes difficult later on to remove them or make them adhere to a specific type

File: src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx [hunk 9]
```diff
@@ -295,9 +298,11 @@ const ExecutionActions: React.FC<ExecutionActionsProps> = props => {
   /*--------------------------------------Retry Pipeline---------------------------------------------*/
 
   /*--------------------------------------Run Pipeline---------------------------------------------*/
-
+  const { CI_YAML_VERSIONING } = useFeatureFlags()
   const reRunPipeline = (): void => {
-    openRunPipelineModal()
+    CI_YAML_VERSIONING && module?.valueOf().toLowerCase() === moduleToModuleNameMapping.ci.toLowerCase()
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx:303]: will cleanup later, after Vardan's changes

File: src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx:82]: right, makes sense

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 10]
```diff
@@ -213,7 +226,13 @@ export function ExecutionHeader({ pipelineMetadata }: ExecutionHeaderProps): Rea
             canExecute={canExecute}
             canRetry={pipelineExecutionSummary.canRetry}
             modules={pipelineExecutionSummary.modules}
-            onReRunInDebugMode={hasCI && CI_REMOTE_DEBUG ? () => openRunPipelineModal() : undefined}
+            onReRunInDebugMode={
+              hasCI && CI_REMOTE_DEBUG
+                ? CI_YAML_VERSIONING && module?.valueOf().toLowerCase() === moduleToModuleNameMapping.ci.toLowerCase()
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:231]: will cleanup later, after Vardan's changes

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetailsHeader/ExecutionStageDetailsHeader.tsx [hunk 11]
```diff
@@ -94,8 +97,11 @@ export function ExecutionStageDetailsHeader(): React.ReactElement {
       ) : null}
     </div>
   )
+  const { CI_YAML_VERSIONING } = useFeatureFlags()
   const runPipeline = (): void => {
-    openRunPipelineModal()
+    CI_YAML_VERSIONING && module?.valueOf().toLowerCase() === moduleToModuleNameMapping.ci.toLowerCase()
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetailsHeader/ExecutionStageDetailsHeader.tsx:102]: will cleanup later, after Vardan's changes

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx [hunk 12]
```diff
@@ -309,10 +313,12 @@ export const MenuCell: CellType = ({ row, column }) => {
     [data.identifier]
   )
 
+  const { CI_YAML_VERSIONING } = useFeatureFlags()
   const runPipeline = (): void => {
-    openRunPipelineModal()
+    CI_YAML_VERSIONING && module?.valueOf().toLowerCase() === moduleToModuleNameMapping.ci.toLowerCase()
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx:318]: will cleanup later, after Vardan's changes

File: src/modules/70-pipeline/v1/components/PipelineCanvasV1/PipelineCanvasV1.tsx [hunk 13]
```diff
@@ -72,10 +70,10 @@ interface PipelineWithGitContextFormProps extends PipelineInfoConfig {
...
 }
 
-interface InputSetValue extends SelectOption {
-  type: InputSetSummaryResponse['inputSetType']
-  gitDetails?: EntityGitDetails
-}
+// interface InputSetValue extends SelectOption {
+//   type: InputSetSummaryResponse['inputSetType']
+//   gitDetails?: EntityGitDetails
+// }
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/PipelineCanvasV1/PipelineCanvasV1.tsx:76]: @vardanbansal-harness this is your code, will delete this but we might need to add it back when run modal supports selection of old input sets

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:196]: sure, makes sense

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:207]: done

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/useRunPipelineModalV1.tsx (hunk 3 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/useRunPipelineModalV1.tsx:110]: yes, wont need this until we make other changes in run modal first

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+      setCodebaseConnector(connectorDetails?.data?.connector)
+    }
+  }, [loadingConnectorDetails, connectorDetails])
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:199]: yes, cleaning it up

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 15]
```diff
@@ -0,0 +1,351 @@
...
+              })
+          } catch (e) {
+            setIsFetchingBranches(false)
+          }
+        }
+      }
+    },
+    [shouldAllowRepoFetch, originalPipeline, codeBaseType]
+  )
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx:260]: done

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/RunPipelineFormV1.tsx (hunk 2 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/RunPipelineFormV1.tsx:257]: yes

File: src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx [hunk 16]
```diff
@@ -0,0 +1,351 @@
...
+                if (result.data?.defaultBranch?.name) {
+                  setIsDefaultBranchSet(true)
+                }
+              })
+              .catch(_e => {
+                setIsFetchingBranches(false)
+              })
+          } catch (e) {
+            setIsFetchingBranches(false)
+          }
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/v1/components/RunPipelineModalV1/CICodebaseInputSetFormV1.tsx]: right, updated

**Review by vkbansal-harness — APPROVED** [review]

**Comment by shaurya-harness** [comment]
retrigger jest

**Review by d1wilko — APPROVED** [review]
lgtm :)

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**